test_scripts/RC/MultipleModules/ModulesAllocation/059_4_GetInteriorVehicleDataConsent_ASK_DRIVER_multiple_modules_cached_and_free.lua

```diff
@@ -0,0 +1,140 @@
+}
+
+local testServiceArea = common.grid.BACK_SEATS
+
+local accessMode = "ASK_DRIVER"
+
+local rcAppIds = { 1, 2 }
+
+local sdlDecisions = {}
+local rcCapabilities = common.initHmiRcCapabilitiesMultiConsent(testServiceArea)
+local testModules = common.buildTestModulesStruct(rcCapabilities)
+
+--[[ Local Variables ]]
+local function driverConsentForReallocationToApp(pAppId, pModuleType, pModuleConsentArray, pRCAppIds, pAccessMode)
+  for moduleId, isAllowed in pairs(pModuleConsentArray) do
+      sdlDecisions[moduleId] = isAllowed  -- SDL makes decision without asking driver for this module
+  end
+  common.driverConsentForReallocationToApp(pAppId, pModuleType, pModuleConsentArray,
+                                           pRCAppIds, pAccessMode, sdlDecisions)
+end
+
+local function driverConsentForReallocationToAppPart(pAppId, pModuleType, pModuleConsentArray, pRCAppIds, pAccessMode)
+  local partialModuleConsentArray = {}
+  for moduleId, isAllowed in pairs(pModuleConsentArray) do
+    if not isAllowed then
+      partialModuleConsentArray[moduleId] = isAllowed
+    end
+    common.driverConsentForReallocationToApp(pAppId, pModuleType, partialModuleConsentArray, pRCAppIds, pAccessMode)
+  end
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Prepare preloaded policy table", common.preparePreloadedPT, { rcAppIds })
+runner.Step("Prepare RC modules capabilities and initial modules data", common.initHmiDataState, { rcCapabilities })
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start, { rcCapabilities })
+runner.Step("Set RA mode: ".. accessMode, common.defineRAMode, { true, accessMode })
+runner.Step("Register App1", common.registerAppWOPTU, { 1 })
+runner.Step("Register App2", common.registerAppWOPTU, { 2 })
+runner.Step("Activate App1", common.activateApp, { 1 })
+runner.Step("Send user location of App1 (Back Seat)", common.setUserLocation, { 1, appLocation[1] })
+runner.Step("Activate App2", common.activateApp, { 2 })
+runner.Step("Send user location of App2 (Back seat)", common.setUserLocation, { 2, appLocation[2] })
+
+for moduleType, consentArray in pairs(testModules) do
+  for moduleId, isAllowed in pairs(consentArray) do
+    if isAllowed == false then
+      runner.Step("Allocate module [" .. moduleType .. ":" .. moduleId .. "] to App2",
+        common.rpcSuccess, { moduleType, moduleId, 2, "SetInteriorVehicleData" })
+    end
+  end
+end
+
+runner.Title("Test")
+runner.Step("Activate App1", common.activateApp, { 1 })
+for moduleType, consentArray in pairs(testModules) do
+  runner.Step("Allow/disallow " .. moduleType .. " modules reallocation to App1 without asking driver",
+    driverConsentForReallocationToAppPart, { 1, moduleType, consentArray, rcAppIds, accessMode })
+end
+
+for moduleType, consentArray in pairs(testModules) do
+  runner.Step("Allow/disallow " .. moduleType .. " modules reallocation to App1 without asking driver",
```

Comment:
This step does get negative consent from driver despite saying `without asking driver`.